core/templates/dev/head/pages/collection_editor/editor_tab/CollectionNodeCreatorDirective.js

```diff
@@ -28,18 +28,53 @@ oppia.directive('collectionNodeCreator', [
         'validatorsService', 'CollectionEditorStateService',
         'CollectionLinearizerService', 'CollectionUpdateService',
         'CollectionNodeObjectFactory', 'ExplorationSummaryBackendApiService',
-        'siteAnalyticsService',
+        'SearchExplorationsBackendApiService', 'siteAnalyticsService',
         function(
             $scope, $http, $window, $filter, alertsService,
             validatorsService, CollectionEditorStateService,
             CollectionLinearizerService, CollectionUpdateService,
             CollectionNodeObjectFactory, ExplorationSummaryBackendApiService,
-            siteAnalyticsService) {
+            SearchExplorationsBackendApiService, siteAnalyticsService) {
           $scope.collection = CollectionEditorStateService.getCollection();
           $scope.newExplorationId = '';
           $scope.newExplorationTitle = '';
+          $scope.searchQueryHasError = false;
+
           var CREATE_NEW_EXPLORATION_URL_TEMPLATE = '/create/<exploration_id>';
 
+          /**
+           * Fetches a list of exploration metadata dicts from backend, given
+           * a search query. It then extracts the title and id of the
+           * exploration to prepare typeahead options.
+           */
+          $scope.fetchExplorationTitleFragment = function(searchQuery) {
```

Comment:
fetchTypeaheadResults, perhaps. I'm not sure what an "exploration title fragment" is.